lib/billy/proxy_connection.rb

```diff
@@ -52,11 +52,9 @@ def on_message_complete
     def restart_with_ssl(url)
       @ssl = url
       @parser = Http::Parser.new(self)
+      generate_certificate_chain(url)
       send_data("HTTP/1.0 200 Connection established\r\nProxy-agent: Puffing-Billy/0.0.0\r\n\r\n")
-      start_tls(
-        private_key_file: File.expand_path('../mitm.key', __FILE__),
-        cert_chain_file: File.expand_path('../mitm.crt', __FILE__)
-      )
+      start_tls(private_key_file: @key_file, cert_chain_file: @chain_file)
```

Comment:
I think we should avoid the use of instance variables.  Perhaps the new helper method can return a hash and be used directly as a param in this method call?

Reply:
Good point.